### ISSUE TITLE
`@stylistic/function-parentheses-newline-inside` fix takes the code behind a function into an inline comment

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#129]
Found while fixing #113, by an adversarial sweep of the rule that fix left. The closing parenthesis is now guarded; the opening one is not, and the same option breaks a file through it. The damage is the same as #113's — a fix takes a line break away and the code behind it ends up inside an inline comment — but it reaches the code through a comment that belongs to another function of the same value, so nothing about the function being fixed betrays it.

### What minimal example or steps are needed to reproduce the bug?

```less
a {
  t: foo(1px // c) calc(
2px);
}
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "customSyntax": "postcss-less",
  "plugins": "@stylistic/stylelint-plugin",
  "rules": {
    "@stylistic/function-parentheses-newline-inside": "never-multi-line"
  }
}
```

### What did you expect to happen?

The line break in front of `2px` is what closes the `// c) calc(` comment, so `never-multi-line` cannot take it away without commenting out everything behind it. The fix should leave the value alone and let the warning stand, as it now does for the closing parenthesis.

### What actually happened?

```less
a {
  t: foo(1px // c) calc(2px);
}
```

Everything from the double slash to the end of the line is comment text now, the declaration ends nowhere, and no warning survives the run: `--fix` reports a clean pass on a file it has just broken.

The cause is at [`lib/rules/function-parentheses-newline-inside/index.js:180-186`](https://github.com/stylelint-stylistic/stylelint-stylistic/blob/main/lib/rules/function-parentheses-newline-inside/index.js#L180-L186), whose `fixBeforeForNever` empties the whitespace the function keeps in front of its first argument without asking what stands in front of that whitespace. Here that whitespace is `calc`'s own `before`, and what stands in front of it is the tail of a comment opened inside `foo`.

### How much is at stake

Little, and the note is worth making: for the input to exist, the comment must already swallow a closing parenthesis of its own, so the file is already unbalanced for less.js and for Sass alike, and only `postcss-less` reads it at all — it keeps a double slash in a value as ordinary text, which is what #32 was about. No valid stylesheet reaches this. What makes it worth an issue rather than a note is #115: the sweep #113, #114, #116 and #117 belong to exists to leave no fixer writing into a comment before the values a syntax prints start being written, and this is the one path of that rule still doing it.

### Do you have a proposal to fix the bug?

The shape the cluster settled on, mirrored: where the text the fix would leave in front of the first argument ends with an inline comment, report the problem and write nothing. `endsWithInlineComment` in `lib/utils/` answers that, and `beforeClosingString` in the same rule already works out what a fix would leave behind — the opening side needs the same, read from the value tree the same way, and asked with the same two polarities.

One thing to settle while doing it: the closing guard is worked out from the value as it stands, while the opening fix rewrites the parsed value in the same pass, so a fix let through on the closing side can have its premise taken away by the opening side afterwards. Neither guard sees the other's writes today. A shape that exercises it needs the same already-broken input, so it is a matter of the two guards being worked out against the same text rather than a defect of its own.

**Comment by firefoxic** [comment]
`function-parentheses-space-inside` has the same path, and it is worth naming here rather than filing twice: all four of its options overwrite `valueNode.before` unguarded, and the same shape breaks the same way.

```less
a {
  t: foo(1px // c) calc(
2px);
}
```

Under `never` it becomes `t: foo(1px // c) calc(2px);` with no warning left, and `always` does the same. Byte for byte the behaviour of `main`, so neither #113 nor #114 introduced it and neither repairs it.

The reading of how much is at stake carries over unchanged: for a comment to stand open in front of a function's own parenthesis, the double slash has to precede it with no line break, so that parenthesis is commented out already and the value is unbalanced before anything is run. What is new is that the single-line options reach it too, since `isSingleLineString` reads a line feed and a carriage return and nothing else while Sass ends an inline comment on a form feed as well: `a { t: foo(1px // c) calc(\f2px); }` is one line to the rule and two to the syntax.

Whoever takes this on should therefore mirror the guard into both rules at once. `beforeClosingString` in either of them already works out what a fix would leave behind and asks it in the two polarities the closing side needs; the opening side needs the same, read from the value tree the same way.
